src/components/Pressable/PressableWithFeedback.js

```diff
@@ -12,28 +13,50 @@ const omittedProps = ['style', 'pressStyle', 'hoverStyle', 'focusStyle', 'wrappe
 const PressableWithFeedbackPropTypes = {
     ..._.omit(GenericPressablePropTypes.pressablePropTypes, omittedProps),
     pressDimmingValue: propTypes.number,
-    hoverDimmingValue: propTypes.number,
 };
 
 const PressableWithFeedbackDefaultProps = {
     ..._.omit(GenericPressablePropTypes.defaultProps, omittedProps),
     pressDimmingValue: variables.pressDimValue,
-    hoverDimmingValue: variables.hoverDimValue,
     wrapperStyle: [],
 };
 
 const PressableWithFeedback = forwardRef((props, ref) => {
     const propsWithoutStyling = _.omit(props, omittedProps);
+    const [disabled, setDisabled] = useState(props.disabled);
+
+    useEffect(() => {
+        setDisabled(props.disabled);
+    }, [props.disabled]);
+
     return (
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        <GenericPressable ref={ref} style={props.wrapperStyle} {...propsWithoutStyling}>
+        <GenericPressable
+            ref={ref}
+            style={props.wrapperStyle}
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...propsWithoutStyling}
+            disabled={disabled}
+            onPress={(e) => {
+                setDisabled(true);
+                const onPress = props.onPress(e);
+                InteractionManager.runAfterInteractions(() => {
+                    if (!(onPress instanceof Promise)) {
+                        setDisabled(props.disabled);
+                        return;
+                    }
+                    onPress.then(() => {
+                        setDisabled(props.disabled);
+                    });
+                });
+            }}
+        >
             {state => (
                 <OpacityView
-                    shouldDim={state.pressed || state.hovered}
-                    dimmingValue={state.pressed ? props.pressDimmingValue : props.hoverDimmingValue}
+                    shouldDim={!disabled && state.pressed}
+                    dimmingValue={props.pressDimmingValue}
                     style={[
                         StyleUtils.parseStyleFromFunction(props.style, state),
-                        state.pressed && StyleUtils.parseStyleFromFunction(props.pressStyle, state),
+                        !disabled && state.pressed && StyleUtils.parseStyleFromFunction(props.pressStyle, state),
                         state.hovered && StyleUtils.parseStyleAsArray(props.hoverStyle, state),
                         state.focused && StyleUtils.parseStyleAsArray(props.focusStyle, state),
```

Comment:
@s77rt @robertKozik do you guys think I should add the `!disabled` check for the below styles too? @shawnborton that would mean no hover or focus effect when the button is disabled which I think makes sense but just need confirmation from you guys